### PR TITLE
Cleanup and remove unused parameters from genCreateAddrMode

### LINKED ISSUE #18177: JIT: Remove unused parameters from genCreateAddrMode
Now that the `LEGACY_BACKEND` has been disabled, this method is always called with:
- `regMask == RBM_NONE`
- `nogen == true`

These parameters should be removed.

In addition, the `scale` parameter is only defined under `SCALED_ADDR_MODES`, however `Lowering` calls it with that parameter unconditionally. Currently, all targets define that to be true, but for now it's probably best to at least modify the call in lower.cpp to be consistent.

Finally, the header comment talks about `LEA_AVAILABLE` when it should be `SCALED_ADDR_MODES`, and it would be good to clean up this header to follow the format here: https://github.com/dotnet/coreclr/blob/master/Documentation/coding-guidelines/clr-jit-coding-conventions.md#94-function-header-comment

### PR DESCRIPTION
**Pull request by Suchiman**
fixes #18177

### CODE REVIEW

File: src/jit/codegencommon.cpp [hunk 1]
```diff
@@ -1269,18 +1269,14 @@ unsigned CodeGenInterface::InferStructOpSizeAlign(GenTree* op, unsigned* alignme
 bool CodeGen::genCreateAddrMode(GenTree*  addr,
                                 int       mode,
                                 bool      fold,
-                                regMaskTP regMask,
                                 bool*     revPtr,
                                 GenTree** rv1Ptr,
                                 GenTree** rv2Ptr,
 #if SCALED_ADDR_MODES
                                 unsigned* mulPtr,
 #endif
```
Comment (sandreenko) [inline src/jit/codegencommon.cpp]: Could you please add `// SCALED_ADDR_MODES ` to `#endif`?
`#endif // SCALED_ADDR_MODES`

These blocks are very small, but it is better to follow jit coding convention. https://github.com/dotnet/coreclr/blob/master/Documentation/coding-guidelines/clr-jit-coding-conventions.md#75-commenting-ifdefs

**Comment by sandreenko** [comment]
To fix formatting errors apply job patch or clone https://github.com/dotnet/jitutils and run `jit-format.bat -c C:\git\coreclr --fix`.

File: src/jit/codegencommon.cpp [hunk 2]
```diff
@@ -1269,18 +1269,14 @@ unsigned CodeGenInterface::InferStructOpSizeAlign(GenTree* op, unsigned* alignme
 bool CodeGen::genCreateAddrMode(GenTree*  addr,
                                 int       mode,
```
Comment (AndyAyersMS) [inline src/jit/codegencommon.cpp]: Seems like `mode` is also not used. Can we get rid of it and the comment chunk it inspires?

File: src/jit/gentree.cpp [hunk 3]
```diff
@@ -3606,18 +3606,15 @@ unsigned Compiler::gtSetEvalOrder(GenTree* tree)
...
+                                                       &rev,  // reverse ops
+                                                       &base, // base addr
+                                                       &idx,  // index val
 #if SCALED_ADDR_MODES
-                                                       &mul, // scaling
-#endif
-                                                       &cns,  // displacement
-                                                       true)) // don't generate code
+                                                       &mul,  // scaling
+#endif                                                        // SCALED_ADDR_MODES
```
Comment (Suchiman) [inline src/jit/gentree.cpp]: @sandreenko running the formatter seems to indent the comment here, is that desired?

Comment (sandreenko) [reply src/jit/gentree.cpp]: Not really, I think the formatter should not do that, but it is very rare case, so  I am ok with it.

**Comment by AndyAyersMS** [comment]
cc @dotnet/jit-contrib

**Comment by CarolEidt** [comment]
> One might question the effectiveness of algorithm but that's a separable issue.

I would consider that part of #5526, but cleaning up the unused code is great, and doesn't require the same amount of re-engineering :smile:

**Pull request merged**